### PR TITLE
Chapter1/homework/3

### PR DESCRIPTION
**Pull request by Dancs777**
Kinda messed up, little bit clueless. Tell me what and how to correct, and will do. Hopefully got some things good.

### CODE REVIEW

**Comment by timk2020** [comment]
Read in #bc-homework for the instructions for homework3. You should create a new class called Lesson3.cs in the BootCamp.Chapter namespace first, then place a static method called Demo() inside of it. Then open your Program.cs and link the Lesson3.Demo(); inside the Main program entry point. The rest you can read on the channel mentioned above. Feel free to ask for more help on the community with regard to this lesson.

**Comment by timk2020** [comment]
Read in **#bc-homework** for the instructions for homework3. Since I cannot find your Lesson3.cs yet, I think you do not know what needs to be done or a little confused. You should not be messing with the **Lesson3Test.cs** file. You should create a new class called **Lesson3.cs** in the **BootCamp.Chapter** namespace first, then place a static method called **Demo()** inside of it. Then open your **Program.cs** and link the **Lesson3.Demo();** inside the Main program entry point. The rest you can read more on the channel mentioned above(**#bc-homework** on Discord). Feel free to ask mentors for more help on the community with regard to your problems.

**Comment by ellmaster** [comment]
This is hard to tell because you want to merge into the wrong branch. Change the base branch to be the correct one or close this PR and open a new one to the correct branch.

File: README.md [hunk 1]
```diff
@@ -1,77 +1,70 @@
...
+  * Click fetch.
+  * Click pull.
 
+## Task
+1) Open the Homework 3:
+  * Choose the Chapter1/Homework/3 branch in the bottom right.
+    * If it is not there already:
+      * Click manage branches.
+      * Open remotes/origin/Chapter1/Homework and rightclick 3 and choose fetch.
+      * There should be a "Chapter1/Homework/3" path be available now. Choose that one.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 58
           

Complexity increasing per file
==============================
- Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs  1
- Src/BootCamp.Chapter/Checks.cs  1
- Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599&bid=16162435)

File: README.md [hunk 2]
```diff
@@ -1,77 +1,70 @@
...
+  ```
+6) Refactor/edit Lesson3.Demo() by using functions.
+  * There should be as little duplicate code as possible
+  * There should be functions for:
+    * Calculating BMI (weight comes in kg, height comes in meters),
+    * Prompt for input and converting it to int (print message for request, read console input and return converted input to int), 
+    * Prompt for input and converting it to string (print message for request, read console input and return input),
+    * Prompt for input and converting it to float (print message for request, read console input and return converted input to float).
+7) Call "Lesson3.X(y)" functions from the Checks.cs file.
+  * X is the name of your function.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 3]
```diff
@@ -1,77 +1,70 @@
...
+    }
+  ```
+6) Refactor/edit Lesson3.Demo() by using functions.
+  * There should be as little duplicate code as possible
+  * There should be functions for:
+    * Calculating BMI (weight comes in kg, height comes in meters),
+    * Prompt for input and converting it to int (print message for request, read console input and return converted input to int), 
+    * Prompt for input and converting it to string (print message for request, read console input and return input),
+    * Prompt for input and converting it to float (print message for request, read console input and return converted input to float).
+7) Call "Lesson3.X(y)" functions from the Checks.cs file.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 4]
```diff
@@ -1,77 +1,70 @@
...
+  * X is the name of your function.
+  * y are the parameters of your function. If you do not have parameters then there is no y.
+8) Run tests, make sure you pass all the tests.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 5]
```diff
@@ -1,77 +1,70 @@
...
+  * Open the solution explorer and Bootcamp.sln (if not already open).
+2) Take homework 2 code, place it in the Program.cs file of the current solution and make sure it still compiles and runs.
+3) Create a file Lesson3.cs:
+  * Rightclick BootCamp.Chapter.
+  * Click add.
+  * Click class.
+  * Type in Lesson3 and hit enter. (You can omit the .cs, it will be put there anyway.)
+4) Write a public static void Demo() function in this file.
+5) Put all Program.cs logic to Lesson3.cs. Call it from Main function.
+  * Do not delete the function or it's curly braces.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 6]
```diff
@@ -1,77 +1,70 @@
...
+  ```csharp
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            Lesson3.Demo();
+        }
+    }
+  ```
+6) Refactor/edit Lesson3.Demo() by using functions.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-content-indent] Don’t use mixed indentation for children, add 4 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 7]
```diff
@@ -1,77 +1,70 @@
...
- 
+## Hints
+* This will make homework 4 much more easy:
+  * Do use Environment.NewLine if you want to concanate a string with a newline command (or use WriteLine).
+  * Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
+* The better your design, the easier it will be to find and fix errors!
+* Sometimes you have to sync new changes from the C# Inn repository:
+  * Click the pencil symbol at the bottom right.
+  * Commit your changes or stash them (after finishing homework 2 there should none be left).
+  * Choose the correct branch in the bottom right (in this case Chapter1/Homework/3 should already be selected).
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 8]
```diff
@@ -1,77 +1,70 @@
...
+  * Click the sync button.
+  * Click fetch.
+  * Click pull.
 
+## Task
+1) Open the Homework 3:
+  * Choose the Chapter1/Homework/3 branch in the bottom right.
+    * If it is not there already:
+      * Click manage branches.
+      * Open remotes/origin/Chapter1/Homework and rightclick 3 and choose fetch.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md (hunk 2 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 9]
```diff
@@ -1,77 +1,70 @@
...
+      * There should be a "Chapter1/Homework/3" path be available now. Choose that one.
+    * If it was there already you should sync as in the hint above.
+  * Open the solution explorer and Bootcamp.sln (if not already open).
+2) Take homework 2 code, place it in the Program.cs file of the current solution and make sure it still compiles and runs.
+3) Create a file Lesson3.cs:
+  * Rightclick BootCamp.Chapter.
+  * Click add.
+  * Click class.
+  * Type in Lesson3 and hit enter. (You can omit the .cs, it will be put there anyway.)
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 10]
```diff
@@ -1,77 +1,70 @@
...
+  * Click pull.
 
+## Task
+1) Open the Homework 3:
+  * Choose the Chapter1/Homework/3 branch in the bottom right.
+    * If it is not there already:
+      * Click manage branches.
+      * Open remotes/origin/Chapter1/Homework and rightclick 3 and choose fetch.
+      * There should be a "Chapter1/Homework/3" path be available now. Choose that one.
+    * If it was there already you should sync as in the hint above.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 11]
```diff
@@ -1,77 +1,70 @@
...
-
-
- 
-
- 
+## Hints
+* This will make homework 4 much more easy:
+  * Do use Environment.NewLine if you want to concanate a string with a newline command (or use WriteLine).
+  * Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
+* The better your design, the easier it will be to find and fix errors!
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs [hunk 12]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 13]
```diff
@@ -1,77 +1,70 @@
...
+      * Click manage branches.
+      * Open remotes/origin/Chapter1/Homework and rightclick 3 and choose fetch.
+      * There should be a "Chapter1/Homework/3" path be available now. Choose that one.
+    * If it was there already you should sync as in the hint above.
+  * Open the solution explorer and Bootcamp.sln (if not already open).
+2) Take homework 2 code, place it in the Program.cs file of the current solution and make sure it still compiles and runs.
+3) Create a file Lesson3.cs:
+  * Rightclick BootCamp.Chapter.
+  * Click add.
+  * Click class.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 14]
```diff
@@ -1,77 +1,70 @@
...
-
- 
+## Hints
+* This will make homework 4 much more easy:
+  * Do use Environment.NewLine if you want to concanate a string with a newline command (or use WriteLine).
+  * Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
+* The better your design, the easier it will be to find and fix errors!
+* Sometimes you have to sync new changes from the C# Inn repository:
+  * Click the pencil symbol at the bottom right.
+  * Commit your changes or stash them (after finishing homework 2 there should none be left).
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 15]
```diff
@@ -1,77 +1,70 @@
...
- 
-
- 
+## Hints
+* This will make homework 4 much more easy:
+  * Do use Environment.NewLine if you want to concanate a string with a newline command (or use WriteLine).
+  * Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
+* The better your design, the easier it will be to find and fix errors!
+* Sometimes you have to sync new changes from the C# Inn repository:
+  * Click the pencil symbol at the bottom right.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md (hunk 1 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: Src/BootCamp.Chapter/Program.cs [hunk 16]
```diff
@@ -1,12 +1,14 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.IO;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 17]
```diff
@@ -1,77 +1,70 @@
...
+  * Commit your changes or stash them (after finishing homework 2 there should none be left).
+  * Choose the correct branch in the bottom right (in this case Chapter1/Homework/3 should already be selected).
+  * Click the house button straight above (upper end).
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 18]
```diff
@@ -1,77 +1,70 @@
...
+3) Create a file Lesson3.cs:
+  * Rightclick BootCamp.Chapter.
+  * Click add.
+  * Click class.
+  * Type in Lesson3 and hit enter. (You can omit the .cs, it will be put there anyway.)
+4) Write a public static void Demo() function in this file.
+5) Put all Program.cs logic to Lesson3.cs. Call it from Main function.
+  * Do not delete the function or it's curly braces.
+  * Write between the curly braces "Lesson3.Demo();" without the ".
+  * This should look like this:
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 19]
```diff
@@ -1,77 +1,70 @@
...
+  * Commit your changes or stash them (after finishing homework 2 there should none be left).
+  * Choose the correct branch in the bottom right (in this case Chapter1/Homework/3 should already be selected).
+  * Click the house button straight above (upper end).
+  * Click the sync button.
+  * Click fetch.
+  * Click pull.
 
+## Task
+1) Open the Homework 3:
+  * Choose the Chapter1/Homework/3 branch in the bottom right.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md (hunk 6 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 20]
```diff
@@ -1,77 +1,70 @@
...
+2) Take homework 2 code, place it in the Program.cs file of the current solution and make sure it still compiles and runs.
+3) Create a file Lesson3.cs:
+  * Rightclick BootCamp.Chapter.
+  * Click add.
+  * Click class.
+  * Type in Lesson3 and hit enter. (You can omit the .cs, it will be put there anyway.)
+4) Write a public static void Demo() function in this file.
+5) Put all Program.cs logic to Lesson3.cs. Call it from Main function.
+  * Do not delete the function or it's curly braces.
+  * Write between the curly braces "Lesson3.Demo();" without the ".
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 21]
```diff
@@ -1,77 +1,70 @@
...
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            Lesson3.Demo();
+        }
+    }
+  ```
+6) Refactor/edit Lesson3.Demo() by using functions.
+  * There should be as little duplicate code as possible
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

File: README.md [hunk 22]
```diff
@@ -1,77 +1,70 @@
...
+    * If it is not there already:
+      * Click manage branches.
+      * Open remotes/origin/Chapter1/Homework and rightclick 3 and choose fetch.
+      * There should be a "Chapter1/Homework/3" path be available now. Choose that one.
+    * If it was there already you should sync as in the hint above.
+  * Open the solution explorer and Bootcamp.sln (if not already open).
+2) Take homework 2 code, place it in the Program.cs file of the current solution and make sure it still compiles and runs.
+3) Create a file Lesson3.cs:
+  * Rightclick BootCamp.Chapter.
+  * Click add.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876599)

**Comment by ellmaster** [comment]
Duplicate of #124 by now.
Closed! :)

**Pull request closed without merging**